servers/src/grin/seed.rs

```diff
@@ -312,7 +312,7 @@ fn listen_for_addrs(
 										}
 									}
 									p2p::Error::PeerWithSelf => break,
-									_ => continue,
+									_ => (),
```

Comment:
Should we also update the peer state to defunct in that case?

Reply:
I have updated it before this `match`:
```
	Err(e) => {
		...
		let _ = peers_c.update_state(addr, p2p::State::Defunct); <<<<  here

		// don't retry if connection refused or PeerWithSelf
		match e {
			p2p::Error::Connection(io_err) => {
				if io::ErrorKind::ConnectionRefused == io_err.kind() {
					break;
				}
			}
			p2p::Error::PeerWithSelf => break,
			_ => (),
		}
	}
```